src/components/Inventory/withdraw/flow/SelectPurposeForm.tsx

```diff
@@ -461,6 +484,57 @@ export default function SelectPurposeForm(props: SelectPurposeFormProps): JSX.El
                   />
                 </Grid>
               )}
+            </Grid>
+            <>
+              {isSingleBatch && !isOutplant && (
+                <>
+                  <Grid display='flex' flexDirection={isMobile ? 'column' : 'row'}>
+                    <Grid item xs={gridSize()} sx={{ marginTop: theme.spacing(2) }} paddingRight={isMobile ? 0 : 1}>
+                      <Textfield
+                        label={strings.NOT_READY_QUANTITY}
+                        id='notReadyQuantityWithdrawn'
+                        onChange={(id: string, value: unknown) => setNotReadyQuantityWithdrawn(value as number)}
+                        type='text'
+                        value={notReadyQuantityWithdrawn}
+                        tooltipTitle={strings.TOOLTIP_NOT_READY_QUANTITY}
+                        className={classes.notReadyQuantityWithdrawn}
+                      />
+                    </Grid>
+                    <Grid item xs={gridSize()} sx={{ marginTop: theme.spacing(2) }} paddingLeft={isMobile ? 0 : 1}>
+                      <DatePicker
+                        id='withdrawnDate'
+                        label={strings.ESTIMATED_READY_DATE}
+                        aria-label={strings.ESTIMATED_READY_DATE}
+                        value={localRecord.readyByDate}
+                        onChange={onChangeDate}
+                      />
+                    </Grid>
+                  </Grid>
+                  <Grid display='flex' flexDirection={isMobile ? 'column' : 'row'}>
+                    <Grid item xs={gridSize()} sx={{ marginTop: theme.spacing(2) }} paddingRight={1}>
+                      <Textfield
+                        label={strings.READY_QUANTITY}
+                        id='readyQuantityWithdrawn'
+                        onChange={(id: string, value: unknown) => setReadyQuantityWithdrawn(value as number)}
+                        type='text'
+                        value={readyQuantityWithdrawn}
+                        tooltipTitle={strings.TOOLTIP_READY_QUANTITY}
+                        className={classes.readyQuantityWithdrawn}
+                      />
+                    </Grid>
+                    <Grid item xs={gridSize()} sx={{ marginTop: theme.spacing(2) }} paddingLeft={1}>
```

Comment:
can we remove the paddingLeft on mobile? looks good otherwise.
btw, see Chudo's comments on validation. Should we do that as a follow up or in this PR?

Reply:
I just added validation here